### ISSUE TITLE
[Feature Request] Add organization stats

### ISSUE THREAD

**Issue by franky47** [anuraghazra/github-readme-stats#1]
Most of my open-source work is moved to dedicated organizations ([47ng](https://github.com/47ng), [FortySevenEffects](https://github.com/FortySevenEffects), [Chiffre](https://github.com/chiffre_io)), for scoping and to avoid cluttering my personal repos.

I'd be happy to contribute a feature to include aggregated organization stats.

API ideas:
```
![Anurag's github stats](https://github-readme-stats.vercel.app/api?username=anuraghazra&orgs=acme,evilcorp,fsociety)
```

**Comment by anuraghazra** [comment]
@franky47 great idea! okay i will look into it. and try to add org support. meanwhile you can also contribute if you want.

**Comment by metaskills** [comment]
I did that in my fork on AWS Lambda. For the `pin.j` I did:

```javascript
      query: `
        fragment RepoInfo on Repository {
          name
          owner {
            login
          }
````

```javascript
const renderRepoCard = (repo) => {
  const owner = repo.owner.login;
```

```javascript
<text x="50" y="38" class="header"><a class="header" href="https://github.com/${owner}/${name}">${name}</a></text>
```

You can see too that I added an `<a>` link to the repo name too.

**Comment by anuraghazra** [comment]
Will it actually show the stats including all the Orgs?

**Comment by anuraghazra** [comment]
@metaskills  Also yes that link would work but will it work in github readmes? Hmmm i dont think soo because it is served as image not as raw html elements. 🤔

**Comment by metaskills** [comment]
Oh yea... good point. Without a <a> in your markdown, you can click it (views svg) then click it again. But easier to put the links in your markdown I suspect.

**Comment by anuraghazra** [comment]
Yes just wrapping the markdown image inside a link is a good way.

**Comment by FairyEver** [comment]
I really hope to have this function, which can count my stars in the organization！

**Comment by deancn** [comment]
my stars value is 0,  can help check?

https://github-readme-stats.vercel.app/api/?username=deancn

**Comment by terror** [comment]
Hi @deancn, it is because we currently only support stars from repositories you are the owner of. Since many people want this as an enhancement I'm sure we'll be able to implement this soon.

**Comment by deancn** [comment]
@terror  got it, thanks.

**Comment by SirWindfield** [comment]
Assuming you might be a member of some huge organization (npm, babel, rust). It might make sense to add a filter on top of it to prevent certain organizations from being included. Better than that would be an allowlist that only includes organizations that you explicitly declare:
`https://github-readme-stats.vercel.app/api/?username=SirWindfield?orgs=zors-engine,Spotifyd`

**Comment by Mr-jiangzhiguo** [comment]
How long this feature could be used?

**Comment by RubenMateus** [comment]
any news on this?

**Comment by ngoldack** [comment]
@anuraghazra I tried it in go and it worked like a charm to count those stars:

```go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

type Org struct {
	Name string `json:"login"`
}

type Repo struct {
	Name  string `json:"full_name"`
	Stars int    `json:"stargazers_count"`
}

func main() {
	url := "https://api.github.com/users/mattetti/orgs"

	resp, err := http.Get(url)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()
	body, err := ioutil.ReadAll(resp.Body)
	if err != nil {
		panic(err)
	}

	var stars int

	var orgs []Org
	json.Unmarshal(body, &orgs)
	for _, v := range orgs {

		url = fmt.Sprintf("https://api.github.com/orgs/%s/repos", v.Name)

		resp, err = http.Get(url)
		if err != nil {
			panic(err)
		}
		defer resp.Body.Close()
		body, err = ioutil.ReadAll(resp.Body)
		if err != nil {
			panic(err)
		}

		var repos []Repo
		json.Unmarshal(body, &repos)

		for _, v := range repos {
			fmt.Println(v.Name, v.Stars)
			stars += v.Stars
		}
	}
	fmt.Println("-----------------------")
	fmt.Println("### Total stars:", stars)
}
```
It's quite simple: you just get all org names from the api endpoint `/users/${USERNAME}/orgs` and then get all each repo from each org from the endpoint `/orgs/${ORGNAME}/repos`. Just count those stars together and you have it. But since I don't have the expierence in javascript I won't be able to implement this.

**Comment by Mr-jiangzhiguo** [comment]
@ngoldack 

this api `/users/${USERNAME}/orgs` seemingly inaccurate
for example `https://api.github.com/users/Mr-jiangzhiguo/orgs` get `[]`, 
but in fact, it has 5 orgs
![image](https://user-images.githubusercontent.com/22312092/91958592-2140c380-ed3a-11ea-827c-9c28cdad9c7f.png)

**Comment by mre** [comment]
@Mr-jiangzhiguo, strangely when I go to your Github profile, I also can't see your organizations. 
I tried mine (`curl https://api.github.com/users/mre/orgs`) and it works. 🤔

**Comment by Mr-jiangzhiguo** [comment]
> @Mr-jiangzhiguo, strangely when I go to your Github profile, I also can't see your organizations.
> I tried mine (`curl https://api.github.com/users/mre/orgs`) and it works. 🤔

I got it, settings is private
![image](https://user-images.githubusercontent.com/22312092/91961872-5e0eb980-ed3e-11ea-84e0-8d59d07b1be2.png)

**Comment by anuraghazra** [comment]
> It's quite simple: you just get all org names from the api endpoint /users/${USERNAME}/orgs and then get all each repo from each org from the endpoint /orgs/${ORGNAME}/repos. Just count those stars together and you have it. But since I don't have the expierence in javascript I won't be able to implement this.

Yeah logic isn't an issue here. we can of course get the data and manipulate it as per our needs. 

BUT! The issue is that vercel serverless execution time is only 10sec and it's not enough to do fetch all those data then manipulate it, serverless function would time out long before we even get to fetching all org data.

> and then get all each repo from each org 
And this is also very expensive because the organization can have many repos and we have to fetch all of them and accumulate the star count.

**Comment by dzikoysk** [comment]
@anuraghazra some kind of lightweight cache could be a solution 🤔

**Comment by ngoldack** [comment]
@anuraghazra or concurrency.

Another problem would be the rate limiting with too much orgs/repos

**Comment by anuraghazra** [comment]
> some kind of lightweight cache could be a solution thinking

We already have caching, it's not the issue since the initial data wont even load because vercel would timeout.

**Comment by zhangyuang** [comment]
the same demands

**Comment by changkun** [comment]
eager to have this.

**Comment by conquerv0** [comment]
Patiently waiting for an exciting update.

**Comment by MrTroble** [comment]
I would prefer to use a whitelist for that, that would also make it more straight forward and light (probably?)

For example:
```https://github-readme-stats.vercel.app/api?username=MrTroble&?orgs=Troblecodings&show_icons=true&theme=radical```

This would go through the repos of Troblecodings and would account for all stars in those. This shouldn't take to long should it?

**Comment by Zethson** [comment]
Still relevant.

**Comment by JKLorenzo** [comment]
Any updates on this?

**Comment by wleoncio** [comment]
Just [bought @anuraghazra a coffee](https://www.buymeacoffee.com/anuraghazra) in the name of this issue (and as a general "thank you" for making this wonderful piece of software). Maybe if more people chip in we can convince him to get this done before he gets a heart attack from all the caffeine. 😅

**Comment by anuraghazra** [comment]
> Just [bought @anuraghazra a coffee](https://www.buymeacoffee.com/anuraghazra) in the name of this issue (and as a general "thank you" for making this wonderful piece of software). Maybe if more people chip in we can convince him to get this done before he gets a heart attack from all the caffeine. 😅

😄 haha thank you for coffee. 

Well yeah probably only viable way to get this issue done is to change the fetching system, like instead of fetching on the fly with serverless functions, we can create a github action to fetch the data that way serverless functions won't timeout and we can fetch more data looping over all the organizations.

**Comment by kaaax0815** [comment]
What is the status of this Feature?

**Comment by franky47** [comment]
The GitHub REST API has an endpoint to list repositories where the [authenticated user is either the owner or a member](https://docs.github.com/en/rest/reference/repos#list-repositories-for-the-authenticated-user), this would make most sense for this kind of stats: if I've pushed a README update to `facebook/react`, its stars should not be credited to my account, but it should if I'm a core member/maintainer.

There are some drawbacks though:
- It's REST, I'm not sure if there's an equivalent in GraphQL (will have to dig it out)
- It's limited to 100 repositories per page, so fetching them all could take a while and bust the 10s Vercel timeout for highly productive users 😅

**Comment by developStorm** [comment]
> @franky47  * It's REST, I'm not sure if there's an equivalent in GraphQL (will have to dig it out)

Does this GraphQL query look like the same thing you're talking about?
https://github.community/t/graphql-api-how-to-fetch-all-repositories-that-the-current-user-has-access-to/13792/10

**Comment by developStorm** [comment]
I was able to use the query params mentioned in my last comment to include org repo for `top-langs` card. Use @franky47's card as demo (since mine was a bit tricky🥲):

(role=OWNER as default)
https://github-readme-stats-one-bice.vercel.app/api/top-langs/?username=franky47&langs_count=10&layout=compact
![](https://github-readme-stats-one-bice.vercel.app/api/top-langs/?username=franky47&langs_count=10&layout=compact)

https://github-readme-stats-one-bice.vercel.app/api/top-langs/?username=franky47&langs_count=10&layout=compact&role=OWNER,ORGANIZATION_MEMBER,COLLABORATOR
![](https://github-readme-stats-one-bice.vercel.app/api/top-langs/?username=franky47&langs_count=10&layout=compact&role=OWNER,ORGANIZATION_MEMBER,COLLABORATOR)


`?role=` could be any combination of `repositoryaffiliation` enum here: https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation

**Comment by wleoncio** [comment]
This looks promising, but when I do mine I get what looks like the language breakdown of _all_ the code in my org repos, not of the code I contributed. I tried other combinations of parameters on that [enum page](https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation) but couldn't get to what I want. 😕

**Comment by franky47** [comment]
Same here: the Eagle code in my demo card was probably 10+ years old example repositories with lots of hardware/PCB files that I wouldn't want shown in my stats, which brings us to filtering again.

**Comment by developStorm** [comment]
> This looks promising, but when I do mine I get what looks like the language breakdown of _all_ the code in my org repos, not of the code I contributed. I tried other combinations of parameters on that [enum page](https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation) but couldn't get to what I want. 😕

That' s right. The API endpoint used here can only get the language distribution for the whole repository but not only for individual's contribution 🤒 It is difficult to do what you say based on the existing queries. 

Since GitHub doesn't seem to have such statistics, I'm worried that the only thing we can do is probably just iterate through all the repositories for all the commits of a given user and calculate the language. And that would be very, very time-consuming - Vercel probably won't do it. I'd be happy to try to implement it if there is a way to get the language distribution data for a single user directly from GitHub API.

**Comment by CircuitSacul** [comment]
Has this been implemented?

**Comment by SirWindfield** [comment]
Can people __please__ stop constantly asking if this has been implemented or not? It doesn't drive development faster, it often puts presure onto the maintainer that gets constantly reminded about this issue (and it's clear from the comments that the solution is not easy). And it notifies people that are subscribed to this thread that are interested in legitimate comments that either give feedback or bring up new ideas on how to solve the issue.

"+1", "is this implemented yet" and similar comments are just unneccessary spam. If the feature would exist, the issue would be closed. It's that easy.

Thank you.

**Comment by developStorm** [comment]
Hey folks! I've made https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-855681098 available for **stars** on general stats card as well. The implementation again used `ownerAffiliations` and you can see detailed explanation of roles you may use [here](https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation).

Use @franky47's stats card as demo (again 😆):

Only include repo they owned:
https://github-readme-stats-one-bice.vercel.app/api?username=franky47&show_icons=true&include_all_commits=true&count_private=true
![](https://github-readme-stats.vercel.app/api?username=franky47&show_icons=true&include_all_commits=true&count_private=true)

Include all repos where they have write access to:
https://github-readme-stats-one-bice.vercel.app/api?username=franky47&show_icons=true&include_all_commits=true&count_private=true&role=OWNER,ORGANIZATION_MEMBER,COLLABORATOR
![](https://github-readme-stats-one-bice.vercel.app/api?username=franky47&show_icons=true&include_all_commits=true&count_private=true&role=OWNER,ORGANIZATION_MEMBER,COLLABORATOR)

\* It's not that accurate to say "have write access to". Check GitHub docs linked above for more precise description.

I'm not sure if this is enough to close this issue -- let me know for any thoughts!

**Comment by CircuitSacul** [comment]
@SirWindfield you're right, sorry about that. My intention wasn't to put any pressure on the developers, I just couldn't tell from the comments whether or not it had been implemented (though I guess I should've noticed the issue was still open).

**Comment by anuraghazra** [comment]
Hey @developStorm this is great thanks for this. 

Although I do have a question, it's a philosophical question matter of fact. 

Like for example, I'm member of [reakit](https://github.com/reakit) organization because I contributed code to it, 
but is it fair to count all of the *4.8k* stars as **MY stars**??? Technically those stars aren't gained by me, those are gained by the reakit organization.

**Comment by developStorm** [comment]
Hi @anuraghazra thanks for the explanation.I think the typical case where `COLLABORATOR` and `ORGANIZATION_MEMBER`· will be used is when some users only use the organization to organize their repo. i.e. The main (if not the only) contributor of these repositories is still the user themselves, even if the repository is under some org. In that case I think it's reasonable for them to say those stars are gained by themselves. This scenario is also mentioned several times in #1.

I don't think it's necessary to overthink anti-cheating as an open source stats project. It's entirely possible for a intended user to create fake data by manually modifying the source code or SVG, etc. - no need to go through such a troublesome as joining/contributing to the organization.

Hope this makes sense ✨

**Comment by rickstaa** [comment]
Here my two cents. @developStorm I think I agree with @anuraghazra that removing the `ORGANIZATION_MEMBER` option would increase the design integrity of the `github-readme-stats` tool. However, as you said, if people want to cheat, they always find a way to do so. Therefore if this requires you to add a lot of code, it might not be worth it since it complicates the codebase. Maybe first let's see what the rest of the userbase/community thinks.

**Comment by rickstaa** [comment]
@developStorm thanks again for creating the pull request. I think it is a very valuable addition to the `github-readme-stats` tool. 

Here my two cents on the matter. I think I agree with @anuraghazra that removing the `ORGANIZATION_MEMBER` option would increase the design integrity of the `github-readme-stats` tool. However, as you said, if people want to cheat, they always find a way to do so. Therefore if this requires you to add a lot of code, it might not be worth it since it complicates the codebase.  Maybe first let's see what the rest of the userbase/community thinks.

**Comment by wleoncio** [comment]
> Maybe first let's see what the rest of the userbase/community thinks.

Well, can't speak for everyone, but I for one care very little about counting stars; I just wish my [programming language breakdown card](https://camo.githubusercontent.com/e80304e0855acb945847de60c4752dc5c339d228358cacf44890e5a87d361234/68747470733a2f2f6769746875622d726561646d652d73746174732e76657263656c2e6170702f6170692f746f702d6c616e67732f3f757365726e616d653d776c656f6e63696f266c616e67735f636f756e743d38266c61796f75743d636f6d7061637426686964655f7469746c653d7472756526686964655f626f726465723d747275652662675f636f6c6f723d666666666666303026746578745f636f6c6f723d363636363636) would reflect all my commits on GH (personal and org repositories; ideally even PR code in 3rd-party repos).

**Comment by anuraghazra** [comment]
@rickstaa agreed. 

I'm fine with adding this `ownerAffiliations` feature to the stats if majority of people thinks it's a useful feature and can provide them with much better stat results. 

Even though I'm still ridgid to my https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-864451245 

I also somewhat agree with @developStorm's thoughts that it's "not necessary to overthink anti-cheating as an open source stats project"

**Comment by rickstaa** [comment]
@wleoncio Excellent point! I forgot that #1122 also influenced the "programming language breakdown card". In that case, based on https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-864459346, I think the benefits of accepting the `ORGANIZATION_MEMBER` param outweigh the prevention of cheating.

**Comment by rickstaa** [comment]
By including the `ORGANIZATION_MEMBER` param. People, can decide themselves which stats they want to show on each card.

My [current setup](https://github.com/rickstaa) is now as follows: I include the `ORGANIZATION_MEMBER` param in my "programming language breakdown card", since it better represents the languages I use daily. However, for the "GitHub stats" card, however, since this would add stars that I would not want to take credit for, I omit it.

**Comment by wleoncio** [comment]
> My [current setup](https://github.com/rickstaa) is now as follows: I include the `ORGANIZATION_MEMBER` param in my "programming language breakdown card", since it better represents the languages I use daily. However, for the "GitHub stats" card, however, since this would add stars that I would not want to take credit for, I omit it.

Sounds like a good solution for now. I'll be doing the same, plus adding `exclude_repo` to filter out org repos that obviously skew my card (basically, projects I am not a major contributor to).

**Comment by developStorm** [comment]
Sorry if this is bothering people on this thread.  

@anuraghazra According to your reply in https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-864858924, in conjunction with the ongoing community interest in this feature (see this discussion and the recent #1035, #1172, #1243), would you consider merging my previously initiated PR #1122, at least as an interim workaround? As the code base changes over time, resolving code conflicts can become more cumbersome, so I would appreciate a final conclusion from you as soon as possible.

**Comment by anuraghazra** [comment]
Hi @developStorm I'll consider your PR this weekend. 
I'll consider it based on the reactions on your comments and majority of them.

**Comment by rickstaa** [comment]
@anuraghazra I added the #1450 to allow people to make their star count more representative when #1122 has been merged. I, for example, am a contributor to some popular repositories for which, based on the number of commits, I don't feel like I deserve the stars. Adding the `exclude_repo` option to the stats card would allow me to exclude these repositories.

**Comment by rickstaa** [comment]
@developStorm Don't users also receive the stars and languages results when they add their user accounts as collaborators of the repositories in the organizations they created (see https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation)? In that case, I have to agree with @anuraghazra https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-864451245 that we should filter out the `ORGANIZATION_MEMBER` value for both the language and stats card.

### Verify

Let's quickly test the assumption above by creating a new organization (i.e. [https://github.com/orgs/github-readme-stats-issue-1-test](https://github.com/orgs/github-readme-stats-issue-1-test/)), creating (i.e. a new repository [github-readme-stats-issue-1-test](https://github.com/github-readme-stats-issue-1-test/github-readme-stats-issue-1-test), adding myself as a member and giving it `2` stars. I received the `2` stars from the organization repository, although I did not commit to it. You can see this behaviour in action [the GraphqExplorer](https://docs.github.com/en/graphql/overview/explorer) by using the following syntax:

```graphq
query userInfo($login: String!, $ownerAffiliations: [RepositoryAffiliation]) {
  user(login: $login) {
    repositories(first: 100, ownerAffiliations: $ownerAffiliations, orderBy: {direction: DESC, field: STARGAZERS}) {
      totalCount
      nodes {
        name
        stargazers {
          totalCount
        }
      }
    }
  }
}
```

while using variables:

```graphq
{"login": "rickstaa", "ownerAffiliations": ["COLLABORATOR","ORGANIZATION_MEMBER"]}
```

**Comment by guillaumearnx** [comment]
Hello,

I've deployed my own Vercel Instance but I'm not able to fetch languages in organizations like with `https://github-readme-stats-one-bice.vercel.app` instance.
I suppose that it's a fork with more features but I'm not able to find the source code. Please can someone help me ?

Thank's

**Comment by rickstaa** [comment]
@guillaumearnx To use the organization feature you have to merge #1122 into one of your branches. If you set up your own Vercel instance according to [the documentation](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance) you should then be able to use the organization feature using the link provided by vercel. 

Feel free to temporary use my Vercel instance until this feature request has been merged. The link for this instance can be found in my personal README.

**Comment by guillaumearnx** [comment]
Thank's @rickstaa. If i want to wetch my private repos, do i need to run my own instance with your fork ? 
Because I can't see my private repos with your instance

**Comment by rickstaa** [comment]
@guillaumearnx In that case, you need to create your own instance since a Personal Access Token needs to be present in the instance. I am happy to help if you run into problems. The following steps should get it working for you:

1. Fork the `https://github.com/anuraghazra/github-readme-stats` repository.
2. Follow the Vercel instance [documentation](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance) to set up your own Vercel instance.
3. Clone the repository to your PC.
4. Create a new branch `git checkout -b org-stats`.
5. Add @developStorm remote to your remotes `git add remote develop git@github.com:developStorm/github-readme-stats.git`.
6. Fetch the remote `git fetch develop`
7. Merge @developStorm changes into your branch `git merge develop/master`.
8. Push these changes to your remote `git push`.
9. On Vercel you will see that an instance has been deployed for the `org-stats` branch.
10. You can use the link that is shown for that deployment to create your cards.

**Comment by guillaumearnx** [comment]
thank's a lot

**Comment by sumchowd** [comment]
While this issue is still open, I found another repository that provides this feature in the short term. Github Trends looks at individual commits as opposed to repository stats, so it counts contributions to open-source repos.

https://www.github.com/avgupta456/github-trends

**Comment by Enaium** [comment]
> While this issue is still open, I found another repository that provides this feature in the short term. Github Trends looks at individual commits as opposed to repository stats, so it counts contributions to open-source repos.
> 
> https://www.github.com/avgupta456/github-trends

but this still can't organization stats

**Comment by KendallDoesCoding** [comment]
Is this coming soon?

**Comment by FLAK-ZOSO** [comment]
@KendallDoesCoding, I really hope that. I've never committed anything in this repository, so I don't even know how much complicated it would be, but I'm sure it would be useful.

**Comment by KendallDoesCoding** [comment]
> @KendallDoesCoding, I really hope that. I've never committed anything in this repository, so I don't even know how much complicated it would be, but I'm sure it would be useful.

Yeah, I want to use it for my organization personally.

**Comment by rickstaa** [comment]
> @developStorm Don't users also receive the stars and languages results when they add their user accounts as collaborators of the repositories in the organizations they created (see https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation)? In that case, I have to agree with @anuraghazra [#1 (comment)](https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-864451245) that we should filter out the `ORGANIZATION_MEMBER` value for both the language and stats card.
> 
> ### Verify
> Let's quickly test the assumption above by creating a new organization (i.e. [https://github.com/orgs/github-readme-stats-issue-1-test](https://github.com/orgs/github-readme-stats-issue-1-test/)), creating (i.e. a new repository [github-readme-stats-issue-1-test](https://github.com/github-readme-stats-issue-1-test/github-readme-stats-issue-1-test), adding myself as a member and giving it `2` stars. I received the `2` stars from the organization repository, although I did not commit to it. You can see this behaviour in action [the GraphqExplorer](https://docs.github.com/en/graphql/overview/explorer) by using the following syntax:
> 
> ```
> query userInfo($login: String!, $ownerAffiliations: [RepositoryAffiliation]) {
>   user(login: $login) {
>     repositories(first: 100, ownerAffiliations: $ownerAffiliations, orderBy: {direction: DESC, field: STARGAZERS}) {
>       totalCount
>       nodes {
>         name
>         stargazers {
>           totalCount
>         }
>       }
>     }
>   }
> }
> ```
> 
> while using variables:
> 
> ```
> {"login": "rickstaa", "ownerAffiliations": ["COLLABORATOR"]}
> ```

@developStorm Friendly, ping to see if you have read my previous comment. 😄

**Comment by developStorm** [comment]
@rickstaa Hi Rick, I appreciate your efforts on this issue! I did see your comment earlier, but I don't see a problem with giving users this freedom of choice. They are the ones who are ultimately responsible for their own user homepage. Since we don't have it as a default, I think we should be okay. 

But then again, after I joined multiple organizations myself, this did get a bit cumbersome to manage. Plus the owners of this stat card project have been very inactive about this long-standing issue (it's been over 150 days since they last promised to review #1122 "this weekend" 😅), I'm a little disheartened about this now. Personally, I would probably choose to migrate to some other GitHub Action-based project to get a greater degree of customizability and longer computation times. If runtime is not an issue, it is well possible to calculate user contributions in a more accurate way (e.g. cloning all repositories and scanning language distribution in commits actual made by the user).

**Comment by joaovian06** [comment]
any update in this issue?

**Comment by rickstaa** [comment]
@joaovian06 Hope my comment above helps! 

A small note for everyone. Altough I understand the importance of this feature, please be aware that @anuraghazra maintains this project in his free time. 🏅 Bumping this issue will therefore not speed up the process but will put stress on the collaborators of this repository. If you want this feature implemented, it is best to show your support using the like buttons under the first comment. If you have questions or suggestions about implementing this feature, feel free to comment on the pull request: https://github.com/anuraghazra/github-readme-stats/pull/1122.

**Comment by developStorm** [comment]
> This, however, favours people that write lengthy (bad) code. 😅 It also neglects all the other things people can do to contribute to an open-source project, like code reviews and answering issues and discussions.

This is true. If you want to achieve absolutely fair distribution, we need to provide a platform for all organization administrators to register and submit the actual workload of their members. Overall, automating the assessment of workload sounds impossible to me. So since you can't do it accurately anyway, you might as well let the users do the job themselves. If a person's home page shows 100,000 stars stats card but no pinned repo shows his contribution to those stars, other people are not fools.

**Comment by VidarDev** [comment]
> 5\. git add remote develop git@github.com:developStorm/github-readme-stats.git

@rickstaa 
fatal: pathspec 'remote' did not match any files <--@phpstorm

**Comment by rickstaa** [comment]
@vi

> > 5. git add remote develop [git@github.com](mailto:git@github.com):developStorm/github-readme-stats.git
> 
> @rickstaa fatal: pathspec 'remote' did not match any files <--@phpstorm

Makes sense. It should be `git remote add develop git@github.com:developStorm/github-readme-stats.git` small typo.

**Comment by VidarDev** [comment]
> > > 5. git add remote develop [git@github.com](mailto:git@github.com):developStorm/github-readme-stats.git
> > 
> > @rickstaa fatal: pathspec 'remote' did not match any files <--@phpstorm
> 
> Makes sense. It should be `git remote add develop git@github.com:developStorm/github-readme-stats.git` small typo. 

I made it easier for myself and foked yours. since I adjusted something and started my vercel instance. 😁

**Comment by PekayTab** [comment]
Any updates on when this will be supported publicly?

**Comment by rickstaa** [comment]
@PekayTab Thanks for your question. As explained in #1122 there are several changes that have to be applied before #1122 can be merged.

> @developStorm Thanks for your detailed response. I have no problem merging this pull request. I have used it for months on my [own GitHub readme](https://www.github.com/rickstaa). 🚀 That's why I have been trying to merge it into the main branch. I also don't care about cheating since, in the end, it is impossible to prevent😋. I only pointed out that:
> 
> 1. I think we should filter out the `ORGANIZATION_MEMBER` group. Having this group enabled will also add stars of repositories in my organizations to which I never committed. I think the `COLLABORATOR` will group will already resolve #1.
> 2. It might be unwise to support `ownerAffiliations` for the language card due to the current implementation of the GraphQL API. The current API and the `ownerAffiliations` option enabled all languages in all repositories a user is a member of will be counted. This might not be a good indication of language fluency. However, if point 1 is fixed, I don't think it will be worse than the incorrect results we currently have for the language card. I think the `exclude_repo` option and #1732 will provide users with more than enough ways to fix their language card if they feel it is incorrect.
>
> As a result, because this feature has been widely requested from the community since  9 Jul 2020, I'm good with merging this PR if the following things are fixed:
> - [ ] https://github.com/anuraghazra/github-readme-stats/pull/1122#discussion_r745574651
> - [ ] https://github.com/anuraghazra/github-readme-stats/pull/1122#discussion_r745574953
> - [ ] The Vercel duration is changed back to `30` seconds.
> 
> Nevertheless, since I'm not the owner of this project, I will leave this decision to @anuraghazra.

_Originally posted by @rickstaa in https://github.com/anuraghazra/github-readme-stats/issues/1122#issuecomment-1152237425_

I'm however waiting for feedback of @anuraghazra before I am planning to apply these changes.

**Comment by mendax1234** [comment]
When will stats for an organization be supported? If so, how to use it?

**Comment by rickstaa** [comment]
Hey @mendax1234, thanks for your interest in this feature. Although it is not officially supported yet, I already use it on my fork (see [my readme](https://github.com/rickstaa)). The steps for using this feature are found [here](https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-981419174). 

Unfortunately, I can not give an ETA for this feature (see https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-1180126339 for more information).

**Comment by RuiGuilherme** [comment]
@rickstaa I just tested your deployment on Vercel with support for organizations, I believe that the "Top Language" is not working correctly. The main languages of the repositories are in Java and that's more than 500 commits and I didn't have any changes in the "Top Language".

Are there any limitations at the moment to check the languages of private organizations?

**Comment by rickstaa** [comment]
@RuiGuilherme If you followed [these steps](https://github.com/anuraghazra/github-readme-stats/issues/1?notification_referrer_id=MDE4Ok5vdGlmaWNhdGlvblRocmVhZDk4NDYzMDg5MToxNzU3MDQzMA%3D%3D#issuecomment-981419174) for deploying your own private instance and added a PAT that has access to the repositories of these private organizations, it should work. The permissions are the same as in the [GraphQL Explorer](https://docs.github.com/en/graphql/overview/explorer) when you're logged in (see https://github.com/anuraghazra/github-readme-stats/discussions/1770#language-card-incorrect).

**Comment by RuiGuilherme** [comment]
> @RuiGuilherme If you followed [these steps](https://github.com/anuraghazra/github-readme-stats/issues/1?notification_referrer_id=MDE4Ok5vdGlmaWNhdGlvblRocmVhZDk4NDYzMDg5MToxNzU3MDQzMA%3D%3D#issuecomment-981419174) for deploying your own private instance and added a PAT that has access to the repositories of these private organizations, it should work. The permissions are the same as in the [GraphQL Explorer](https://docs.github.com/en/graphql/overview/explorer) when you're logged in (see [#1770 (comment)](https://github.com/anuraghazra/github-readme-stats/discussions/1770#language-card-incorrect)).

It worked perfectly, thank you very much for the quick response.

**Comment by VianaSamuel** [comment]
> @guillaumearnx In that case, you need to create your own instance since a Personal Access Token needs to be present in the instance. I am happy to help if you run into problems. The following steps should get it working for you:
> 
> 1. Fork the `https://github.com/anuraghazra/github-readme-stats` repository.
> 2. Follow the Vercel instance [documentation](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance) to set up your own Vercel instance.
> 3. Clone your fork to your PC `git clone <FORK_SSH_URL>`.
> 4. Create a new branch `git checkout -b org-stats`.
> 5. Add @developStorm remote to your remotes `git remote add develop git@github.com:developStorm/github-readme-stats.git`.
> 6. Fetch the remote `git fetch develop`
> 7. Merge @developStorm changes into your branch `git merge develop/master`.
> 8. Push these changes to your remote `git push`.
> 9. On Vercel you will see that an instance has been deployed for the `org-stats` branch.
> 10. You can use the link that is shown for that deployment to create your cards.

So, I'm kinda a newbie on git and I was trying to follow your tutorial.
Everything went fine until the 8th step (```git push```).

That's what I got when I tried to do that in the terminal:
```
fatal: The current branch org-stats has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin org-stats

To have this happen automatically for branches without a tracking
upstream, see 'push.autoSetupRemote' in 'git help config'.
```

So I tried the ```git push --set-upstream origin org-stats``` and everything (apparently) went fine, the Vercel deployed again, but my stats cards was still the same.

If possible, could you give me some help? And apologies in advance for the bothering, I'm completely layman with git.

**Comment by rickstaa** [comment]
> > @guillaumearnx In that case, you need to create your own instance since a Personal Access Token needs to be present in the instance. I am happy to help if you run into problems. The following steps should get it working for you:
> > 
> > 1. Fork the `https://github.com/anuraghazra/github-readme-stats` repository.
> > 2. Follow the Vercel instance [documentation](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance) to set up your own Vercel instance.
> > 3. Clone your fork to your PC `git clone <FORK_SSH_URL>`.
> > 4. Create a new branch `git checkout -b org-stats`.
> > 5. Add @developStorm remote to your remotes `git remote add develop git@github.com:developStorm/github-readme-stats.git`.
> > 6. Fetch the remote `git fetch develop`
> > 7. Merge @developStorm changes into your branch `git merge develop/master`.
> > 8. Push these changes to your remote `git push`.
> > 9. On Vercel you will see that an instance has been deployed for the `org-stats` branch.
> > 10. You can use the link that is shown for that deployment to create your cards.
> 
> So, I'm kinda a newbie on git and I was trying to follow your tutorial. Everything went fine until the 8th step (`git push`).
> 
> That's what I got when I tried to do that in the terminal:
> 
> ```
> fatal: The current branch org-stats has no upstream branch.
> To push the current branch and set the remote as upstream, use
> 
>     git push --set-upstream origin org-stats
> 
> To have this happen automatically for branches without a tracking
> upstream, see 'push.autoSetupRemote' in 'git help config'.
> ```
> 
> So I tried the `git push --set-upstream origin org-stats` and everything (apparently) went fine, the Vercel deployed again, but my stats cards was still the same.
> 
> If possible, could you give me some help? And apologies in advance for the bothering, I'm completely layman with git.

For now, you should add the `&role=OWNER,COLLABORATOR` parameter to your card for the organization results to show up. This is not yet documented since this feature has not yet been merged into the master branch. I will probably change it to `&include_orgs` when we release it.

**Comment by brunolnetto** [comment]
The feature seems mature at this level. Who and when will we merge it on main repository?

**Comment by brunolnetto** [comment]
I followed the instructions [available on the post](https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-1279712669), as you can see by commits and action approval on [post](https://github.com/brunolnetto/github-org-readme-stats). I had to additionally:

1. fix conflicts with master branch;
2. Run command `npm run install && npm run prepare && npm run install prettier && npm run install prettier:check` to assure the local environment was similar to this library's author. 

After auto-deploy my own vercel app and substitution on original repository, it rendered the image below. What did I wrong? :-(

![image](https://user-images.githubusercontent.com/13961685/200127426-1e65798e-6d16-44ca-84f5-caf64cc4d8e0.png)

**Comment by ningding97** [comment]
> @guillaumearnx In that case, you need to create your own instance since a Personal Access Token needs to be present in the instance. I am happy to help if you run into problems. The following steps should get it working for you:
> 
> 1. Fork the `https://github.com/anuraghazra/github-readme-stats` repository.
> 2. Follow the Vercel instance [documentation](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance) to set up your own Vercel instance.
> 3. Clone your fork to your PC `git clone <FORK_SSH_URL>`.
> 4. Create a new branch `git checkout -b org-stats`.
> 5. Add @developStorm remote to your remotes `git remote add develop git@github.com:developStorm/github-readme-stats.git`.
> 6. Fetch the remote `git fetch develop`
> 7. Merge @developStorm changes into your branch `git merge develop/master`.
> 8. Push these changes to your remote `git push`.
> 9. On Vercel you will see that an instance has been deployed for the `org-stats` branch.
> 10. You can use the link that is shown for that deployment to create your cards.

Hi, like many other developers, I maintain open-source projects in an organization but not a personal account. After strictly following the instructions, I find that the number of total stars is still not changed (while the number of total commits increases). I'm not an expert on js, so could you please summarize the current criterion here? For example, if I am a member of an organization, and I have created, admined, and maintained some projects, will the stars of these projects be counted? Thx!

**Comment by LucBerge** [comment]
Hello,
Most of my personnal work has been moved to organizations as well. I'm also really interested about this feature !

**Comment by brunolnetto** [comment]
@LucBerge I failed to reproduce the same as @rickstaa did. So, I used his vercel deployed app :-P. You can use my README content and replace property `username` right-hand side by yours.

**Comment by brunolnetto** [comment]
I think the language count in "Jupyter Notebook" is overrated: [look at my profile](https://github.com/brunolnetto)! This language is much bigger than others, which makes me think it counts code rows instead of files. However, since this programming language uses cells to do its stuff, most file rows are render and styling.

**Comment by wleoncio** [comment]
IIRC the stats already count lines of code, not files. The issue is that behind the simplest Jupyter Notebook is hundreds, maybe thousands of lines of JSON code. For example, I just started and saved an empty ipynb file. This is what I see when I open that same file on vim:

```JSON
{
  "cells": [
   {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {},
    "outputs": [],
    "source": []
   }
  ],
  "metadata": {
   "language_info": {
    "name": "python"
   },
   "orig_nbformat": 4
  },
  "nbformat": 4,
  "nbformat_minor": 2
}
```

That's 19 lines of code for a file containing nothing but an empty cell, so it's no wonder having a couple of Jupyter Notebooks would be enough for it to dominate someone's stats. Until that gets fixed to reflect actual human-written code, you might want to configure your stats to ignore the language.

**Comment by rickstaa** [comment]
@wleoncio Thanks for the detailed explanation. @brunolnetto, you can use the `hide` parameter (i.e. `&hide=jupyter%20notebook`) to remove the jupyter notebook results. See <https://github.com/anuraghazra/github-readme-stats#hiding-individual-stats> for more information.

**Comment by Ran-Xing** [comment]
+1+1

**Comment by rickstaa** [comment]
As of today, people can also use the https://github.com/anuraghazra/github-readme-stats/tree/master_orgs branch in their own Vercel deployments to include organization stats. Please remember that this is an unreleased feature (see #2277), so things can still break. Further, please be aware that because of https://github.com/anuraghazra/github-readme-stats/issues/1852, only the first 100 repositories are used. Including organization, stats might therefore skew your results.

**Comment by rickstaa** [comment]
You can also already use this feature as follows:

[![Anurag's GitHub stats](https://github-readme-stats-git-masterorgs-github-readme-stats-team.vercel.app/api?username=rickstaa&include_orgs)](https://github.com/anuraghazra/github-readme-stats)

```md
[![Anurag's GitHub stats](https://github-readme-stats-git-masterorgs-github-readme-stats-team.vercel.app/api?username=rickstaa&include_orgs)](https://github.com/anuraghazra/github-readme-stats)
```

Please, be aware that since this feature is not yet merged, the exact syntax is changed, so it is better to fork the repository and use the `master_orgs` branch.

**Comment by brunolnetto** [comment]
Great! 😱

**Comment by rickstaa** [comment]
> Great! scream

No problem. As stated in https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-1321887994 I can not give any guarantees that the syntax stays the same since https://github.com/anuraghazra/github-readme-stats/pull/2277 is still under review. It would be best if you used the `master_orgs` branch of your own Vercel deployment.

**Comment by Ran-Xing** [comment]
请尽快安排，我超喜欢，真的牛逼

Please arrange it as soon as possible. I like it very much. NIU BI

**Comment by brunolnetto** [comment]
@rickstaa  I copy-pasted your readme. It means, as soon as you do not update your vercel app instance, I am good. However, since it seems you are the most active collaborator to this project, I must somewhen in the future consider to build my own instance.

**Comment by brunolnetto** [comment]
I earned an A++. I am feeling lovely

**Comment by rickstaa** [comment]
@brunolnetto I think it is best for now that you switch to `https://github-readme-stats-git-masterorgs-github-readme-stats-team.vercel.app` since I'm not updating the `master-rstaa` branch anymore and I might remove it in the future. I'm now using my `master_rstaa` branch.

**Comment by brunolnetto** [comment]
You may erase branch `master_rstaa`. Roger!

**Comment by brunolnetto** [comment]
@rickstaa It will sound neatly picky, but whatever: 

1. Type `brunolnetto` on field `Username`: https://awesome-github-stats.azurewebsites.net/ ;
2. Go to my profile on github: https://github.com/brunolnetto

You may notice the number of stars is not equal. It makes me wonder the reason, but I am not the right person to solve this issue.

**Comment by rickstaa** [comment]
> @rickstaa It will sound neatly picky, but whatever:
> 
> 1. Type `brunolnetto` on field `Username`: https://awesome-github-stats.azurewebsites.net/ ;
> 2. Go to my profile on github: https://github.com/brunolnetto
> 
> You may notice the number of stars is not equal. It makes me wonder the reason, but I am not the right person to solve this issue.

I checked your code and the answer to your question is easy. The tool you gave, https://github.com/brunobritodev/awesome-github-stats, uses the `ORGANIZATION_MEMBER` RepositoryAffiliation role following  (see https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation). I only use the `COLLABORATOR` and `OWNER` roles since I don't think people should receive stars of organization repositories to which they did not contribute.

**Comment by brunolnetto** [comment]
You use your cleverness wisely.

**Comment by TonyTroeff** [comment]
I am owner of the [TryAtSoftware](https://github.com/TryAtSoftware) organization and I can assure you that stats for the repositories within this organization are not included.

**Comment by rickstaa** [comment]
> I am owner of the [TryAtSoftware](https://github.com/TryAtSoftware) organization and I can assure you that stats for the repositories within this organization are not included.

@TonyTroeff, thanks for your question. This feature request (i.e. #1) is about including stats from organization repositories for users that have contributed to these repositories. This feature has not yet been released but can be used as explained in https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-1329809068. Your stats will be included if you are a collaborator to such a repository when setting the `include_orgs` variable.

The feature you are asking for, which I presume is organisation/team stats, is tracked in #654. You can show your support there. If enough people show interest in such a card, it will come in our top 10 feature requests, and I will take a look (see https://github.com/anuraghazra/github-readme-stats/issues/1935).

**Comment by TonyTroeff** [comment]
@rickstaa, i support the idea of #654, but my problem here is different. I own the organization and I have been contributing to all repositories contained there as well (actually I am the only member of the organization and the only contributor until this very moment).
However, if I include the `include_orgs` variable, I do not see the desired changes in my stats

**Comment by rickstaa** [comment]
> @rickstaa, i support the idea of #654, but my problem here is different. I own the organization and I have been contributing to all repositories contained there as well (actually I am the only member of the organization and the only contributor until this very moment). However, if I include the `include_orgs` variable, I do not see the desired changes in my stats

I see, in that case, the `include_orgs` variable that is included in the https://github.com/anuraghazra/github-readme-stats/tree/master_orgs branch should show the stats of the contributions you made to your organizations. 🤔 Please make sure that you are on the right branch. 

The `include_orgs` tag currently works as follows. When enabled, it adds the `COLLABORATOR` `repositoryAffiliation` filter to the GraphQL query:

https://github.com/anuraghazra/github-readme-stats/blob/06aee4f12c1f8675d961405d78e841cad40a25ef/src/fetchers/stats-fetcher.js#L25

As explained in [GitHubs' GraphQL documentation](https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation), this will include repositories of the requested user that are found in organizations. I checked your account using the [GraphQL explorer](https://docs.github.com/en/graphql/overview/explorer)

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    name
    login
    repositories(
      first: 100
      ownerAffiliations: [OWNER, COLLABORATOR]
      orderBy: {direction: DESC, field: STARGAZERS}
    ) {
      nodes {
        name
        stargazers {
          totalCount
        }
      }
      pageInfo {
        hasNextPage
        endCursor
      }
    }
  }
}
```

```graphql
{
  "login": "rickstaa"
}
```

and it looks like the data only shows up when I also add the `ORGANIZATION_MEMBER` repository affiliation flag. I however don't think adding this flag is wise since it will include stats from all repositories of organizations a user is in. Maybe you can contact [GitHub support](https://support.github.com/) and explain that you are trying to get the data of repositories you contributed to in an organization you created, but it is not showing your results. It looks like this is a flaw in their API.

**Comment by brunolnetto** [comment]
@rickstaa I had an idea that might interest us:

Statement: given repositories with tags `COLLABORATOR` and `OWNER`, provide a JSON with `(key, value)` equal to `(id, repository_name)`.

**Comment by rickstaa** [comment]
> @rickstaa I had an idea that might interest us:
> 
> Statement: given repositories with tags `COLLABORATOR` and `OWNER`, provide a JSON with `(key, value)` equal to `(username, [repository_names])`.

Hey @brunolnetto, I'm not sure if I follow you. Do you want to request a feature on [the GitHub feature request board](https://github.com/community/community/discussions) that might help GRS?

**Comment by brunolnetto** [comment]
Great idea! [Done!](https://github.com/community/community/discussions/41361)

**Comment by Sekomer** [comment]
I can't see stats from my private included organization repositories. I currently have my Vercel instance, what should I do to include those stats too? 
thanks

**Comment by BroFromSpace** [comment]
Hello guys. Could you help me please, I have fork repo run it on vercel in master_orgs branch, however stats from organisation I am working in are not displayed. What could the problem be?

**Comment by brunolnetto** [comment]
I recommend you to use @rickstaa 's vercel instance until next release. You can check README.md content on either `/rickstaa/rickstaa` or `/brunolnetto/brunolnetto`.

**Comment by BroFromSpace** [comment]
Unfortunately no changes. Is it required to have some settings in github or in organisation?

**Comment by brunolnetto** [comment]
You have to copy the vercel URL on README.md of above-mentioned repositories and change the property `rickstaa` or `brunolnetto` to yours.

**Comment by BroFromSpace** [comment]
You mean it should be like this `https://github-readme-stats-git-masterrstaa-rickstaa.vercel.app/api?username=brofromspace&show_icons=true&include_all_commits=true&count_private=true&include_orgs=true&locale=en`?

**Comment by brunolnetto** [comment]
It is.

Kyrylo Korablov ***@***.***> schrieb am Mo., 19. Dez. 2022,
12:36:

**Comment by BroFromSpace** [comment]
Yep, but its not working, I see my stats, org stats seems not included.

**Comment by brunolnetto** [comment]
It occurs oftentimes if you belong to some private organization. There are criteria on organization star collection here. You may fit on one of those.

**Comment by brunolnetto** [comment]
I recommend you to read this post comment and its previous: 
https://github.com/anuraghazra/github-readme-stats/issues/1#issuecomment-1329809068

**Comment by brunolnetto** [comment]
@rickstaa it is me again. Now and then, my [profile card](https://github.com/brunolnetto/brunolnetto) with profile activity renders broken. Do you have any idea for the reason? In case the feature is official and there is some available common app, may you give us the reference? Thanks, peace.

**Comment by rickstaa** [comment]
> @rickstaa it is me again. Now and then, my [profile card](https://github.com/brunolnetto/brunolnetto) with profile activity renders broken. Do you have any idea for the reason? In case the feature is official and there is some available common app, may you give us the reference? Thanks, peace.

@brunolnetto no problem. The card is not broken, but it seems the Vercel function that creates it reaches the GitHub CDN timeouts. The card will, therefore, not show up the first time a user visits your page after the Cache is expired. It, however, shows up for the next user or if you refresh the page. https://github.com/anuraghazra/github-readme-stats/pull/2159 might elevate the problem, but the best solution would be to change to a GitHub action (see https://github.com/anuraghazra/github-readme-stats/issues/2179). I, however currently sadly do not have the time to implement it.

**Comment by brunolnetto** [comment]
I see. I had this guess in mind because of the refreshing experience on second attempt. You nailed it clearly without my comemnts. 👏🏽

**Comment by brunolnetto** [comment]
Your last comment on given issues depends on codebase conversion to typescript. Friendly question: is there any (good) progress in this direction?

**Comment by rickstaa** [comment]
> Your last comment on given issues depends on codebase conversion to typescript. Friendly question: is there any (good) progress in this direction?

The truth is that I'm currently focusing on my Master's Thesis, so I currently don't have the time to implement these features 😅. My plan was to look at them after my graduation but feel free to work on them if you want. The most up-to-date PR can be found at https://github.com/anuraghazra/github-readme-stats/pull/2108. A lot has already been done by @HwangTaehyun during the hackathon, but some files still need to be translated to typescript, and some changes that were requested by @anuraghazra still need to be applied. I forgot the exact details of why I wanted the codebase to be translated to TS before implementing https://github.com/anuraghazra/github-readme-stats/pull/2159, but it had something to do with some things requiring a lot more work in plane JS 🤔.

**Comment by brunolnetto** [comment]
Save space for your thesis compilation. :-P 

There are familiarization steps before I can put my hands on any feature development. I will take a look at it, but I cannot promise anything: do not await 😅 .

**Comment by rickstaa** [comment]
> Save space for your thesis compilation. :-P
> 
> There are familiarization steps before I can put my hands on any feature development. I will take a look at it, but I cannot promise anything: do not await sweat_smile .

Haha, no problem. Don't feel pressured it has become quite a codebase over the years 😅. If you have some questions feel free to tag me. I'm planning to do it eventually since the problems also occur on my profile, but now I need to headspace for my studies. 👍🏻

**Comment by brunolnetto** [comment]
There is the app `headspace` already at digital stores. I think you will have to use free-mind instead. hehe

**Comment by rickstaa** [comment]
> There is the app `headspace` already at digital stores. I think you will have to use `freemind` instead. hehe

Haha, I indeed own that app!🤣 Calm is also a good one 🧘🏻‍♂️.

**Comment by rickstaa** [comment]
> > I was able to use the query params mentioned in my last comment to include org repo for `top-langs` card. Use @franky47's card as demo (since mine was a bit tricky🥲):
> > (role=OWNER as default) https://github-readme-stats-one-bice.vercel.app/api/top-langs/?username=franky47&langs_count=10&layout=compact ![](https://camo.githubusercontent.com/a8361be1f142f998a4a5ace8d60f72bf6ff408537ece5be0b0460bd7bf1393b4/68747470733a2f2f6769746875622d726561646d652d73746174732d6f6e652d626963652e76657263656c2e6170702f6170692f746f702d6c616e67732f3f757365726e616d653d6672616e6b793437266c616e67735f636f756e743d3130266c61796f75743d636f6d70616374)
> > https://github-readme-stats-one-bice.vercel.app/api/top-langs/?username=franky47&langs_count=10&layout=compact&role=OWNER,ORGANIZATION_MEMBER,COLLABORATOR ![](https://camo.githubusercontent.com/b365baa36f0654a8277cffa7da2c71b9f81d6a5ea9703256d8b02c707da8d6e9/68747470733a2f2f6769746875622d726561646d652d73746174732d6f6e652d626963652e76657263656c2e6170702f6170692f746f702d6c616e67732f3f757365726e616d653d6672616e6b793437266c616e67735f636f756e743d3130266c61796f75743d636f6d7061637426726f6c653d4f574e45522c4f5247414e495a4154494f4e5f4d454d4245522c434f4c4c41424f5241544f52)
> > `?role=` could be any combination of `repositoryaffiliation` enum here: https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation
> 
> **The bad thing is that when you're using the "COLLABORATOR", it basically includes whether or not you've contributed to someone else's repository.**
> 
> * See in: https://docs.github.com/en/graphql/reference/enums#repositoryaffiliation
> 
> **I had to use exclude_repo to not affect the language progress count, you could also hide the language. It would be good if you could use other parameters, which put this condition, either by commit or interaction!**

@JacquelinBB, thanks for your comment. We are aware of the problems, but this is currently due to the upstream GraphQL API limitation. This problem and more are discussed at https://github.com/anuraghazra/github-readme-stats/issues/1801.

**Comment by BroFromSpace** [comment]
Hi guys, is there any way to include mine PR's and commits from organisation repos to stats?

**Comment by mpl1018** [comment]
We made this based on this issue: https://pullpo.io/products/devcard
[![Pullpo - DevCard](https://devcard.pullpo.io/api?user_id=cfutd7di09ls73b6c2p0&custom_title=Marco%20Pati%C3%B1o%20%40%20Pullpo&custom_subtitle=Founder%20%26%20Developer&show_icons=true&disable_animations=false&title_color=fc418d&text_color=a9fef7&icon_color=f8d847&ring_color=fc418d&bg_color=0,141321,3d3b4a&image_url=https%3A%2F%2Flh3.googleusercontent.com%2Fu%2F1%2Fdrive-viewer%2FAITFw-zuVrfjr9pKtrOoaj7k3Ss-FL86JhYrkn-ClXhNMt84XICt27AFwB9FCIIt80d_Ib2T06h4IB513SwT_FRCa1Ny_zkydQ%3Dw1920-h918)](https://pullpo.io/products/devcard)

**Comment by joaovian06** [comment]
I tried this @mpl1018, but can´t get any values to my devcard. :(

**Comment by brunolnetto** [comment]
@mpl1018 I clicked the button "Create my DevCard" and others. It tries to log me in on Slack. It makes no sense to me.

**Comment by mpl1018** [comment]
UPDATE: 
You should now be able to log in with GitHub and install Pullpo for your desired GitHub organization. Your devcard will be available at https://pullpo.io/app/profile. Sorry for the inconveniences 🙏

Don't worry if your stats appear as zeros; it can take some time to fetch historical data from the organization.

If you want to read about the permissions we ask for, you can check it out at https://docs.pullpo.io/github-permissions.
Additionally, here is a tutorial on how to set up the devcard: https://docs.pullpo.io/devcard-requirements.
Also feel free to reach out to me at marco@pullpo.io

@brunolnetto @erickskrauch @ketrab2004 @joaovian06 @kavanase

**Comment by rickstaa** [comment]
> UPDATE: You should now be able to log in with GitHub and install Pullpo for your desired GitHub organization. Your devcard will be available at https://pullpo.io/app/profile. Sorry for the inconveniences pray
> 
> Don't worry if your stats appear as zeros; it can take some time to fetch historical data from the organization.
> 
> If you want to read about the permissions we ask for, you can check it out at https://docs.pullpo.io/github-permissions. Additionally, here is a tutorial on how to set up the devcard: https://docs.pullpo.io/devcard-requirements. Also feel free to reach out to me at [marco@pullpo.io](mailto:marco@pullpo.io)
> 
> @brunolnetto @erickskrauch @ketrab2004 @joaovian06 @kavanase

@mpl1018, thanks for pointing me to your tool. More platforms to choose from for users is okay. Another provider is https://quine.sh/. Having that said, people can already include organization stats by [deploying their own instance](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own) and merging https://github.com/anuraghazra/github-readme-stats/pull/2459. Furthermore, we will release a GitHub version of GRS after this summer that will give users the ability to get more accurate language and stats results, including organization stats (see https://github.com/anuraghazra/github-readme-stats/issues/2179) 🚀.

**Comment by brunolnetto** [comment]
@mpl1018 I would really like to know how you do such convolute software systems. Really! I am unemployed and feel very inadequate not knowing this.

**Comment by rickstaa** [comment]
> Comment: @mpl1018 I would really like to know how you do such convolute software systems. Really! I am unemployed and feel very inadequate not knowing this.
> 
> Report : I got stuck on this "alert" pop-up component, which does not work (It may be something on its callback, I guess.)
> 
> ![image](https://user-images.githubusercontent.com/13961685/255169898-768a003b-de4e-459f-a759-a7506794a386.png)

I do not want to be a Grinch here, but can we keep the pullpo bug reporting through the site intercom before it becomes a big thread 😅.

![image](https://github.com/anuraghazra/github-readme-stats/assets/17570430/29d67314-3be0-4346-8795-ba237a3218be)

**Comment by brunolnetto** [comment]
This is all I hear: grinch, grinch, grinch.

**Comment by rickstaa** [comment]
> This is all I hear: grinch, grinch, grinch.

🤣

![grinch-coffee](https://github.com/anuraghazra/github-readme-stats/assets/17570430/a6f5c708-0df0-42b4-952c-844a2c1639c7)

**Comment by rickstaa** [comment]
@mpl1018 feel free to answer @francois-rozet's last question here. Just trying to prevent this from becoming a pullpo debug issue since a lot of people are subscribed 😅 .

**Comment by francois-rozet** [comment]
Hey, no problem, but I don't even know why I'm mentioned here :sweat_smile: I guess it is a typo between my handle and the OP's handle.

**Comment by rickstaa** [comment]
> Hey, no problem, but I don't even know why I'm mentioned here sweat_smile I guess it is a typo between my handle and the OP's handle.

Oh, sorry for the noise 😅. I should not try to answer GitHub issues quickly before shutting down at the end of the day. Should have been @brunolnetto indeed 👍🏻.

**Comment by brunolnetto** [comment]
I make this inconvenience sometimes, guys. Broad context makes me think in general answer anywwhere. Good job anyway.

**Comment by AsakusaRinne** [comment]
If a repository is created by me in a organization, and I'm a only a member instead of owner of this organization, it seems not to work. Anyone has an idea about it?

**Comment by qwerty541** [comment]
> If a repository is created by me in a organization, and I'm a only a member instead of owner of this organization, it seems not to work. Anyone has an idea about it?

Hey, @AsakusaRinne! Do I understand correctly that you have deployed that branch https://github.com/anuraghazra/github-readme-stats/pull/1122 on own instance, since if not this feature should not work because master branch does not contains that changes. In case if correct branch is deployed, the following values should be available to be passed into `role` querystring parameter: `OWNER`, `COLLABORATOR`, `ORGANIZATION_MEMBER`. I hope that one of them should work, also you are able to pass both this roles at once separating them with commas. If no role or combination of roles works, then this branch really needs fixing.

**Comment by AsakusaRinne** [comment]
> > If a repository is created by me in a organization, and I'm a only a member instead of owner of this organization, it seems not to work. Anyone has an idea about it?
> 
> Hey, @AsakusaRinne! Do I understand correctly that you have deployed that branch #1122 on own instance, since if not this feature should not work because master branch does not contains that changes. In case if correct branch is deployed, the following values should be available to be passed into `role` querystring parameter: `OWNER`, `COLLABORATOR`, `ORGANIZATION_MEMBER`. I hope that one of them should work, also you are able to pass both this roles at once separating them with commas. If no role or combination of roles works, then this branch really needs fixing.

Hi, sorry for the disturbing. I mistook the answers in this issue as the feature (`role` querystring parameter) has already supported by the current master branch. I've succeeded to get what I want today. Thank you so much!

**Comment by pplmx** [comment]
Any progress?

**Comment by ducquoc** [comment]
> Any progress?

Please! It's more than 4 years this ticket is "Open" !

**Comment by wail-asad** [comment]
"I have merged [this](https://github.com/anuraghazra/github-readme-stats/pull/2459) pull request and resolved the conflicts in my [forked repository](https://github.com/wail-asad/github-readme-stats). You can deploy it Vercel like [this video](https://www.youtube.com/watch?v=n6d4KHSKqGk&t=107s) and add the following parameters to the links in your README: &role=OWNER,COLLABORATOR,ORGANIZATION_MEMBER."

**Comment by florianpfleiderer** [comment]
> I have merged [this](https://github.com/anuraghazra/github-readme-stats/pull/2459) pull request and resolved the conflicts in my [forked repository](https://github.com/wail-asad/github-readme-stats). You can deploy it Vercel like [this video](https://www.youtube.com/watch?v=n6d4KHSKqGk&t=107s) and add the following parameters to the links in your README: &role=OWNER,COLLABORATOR,ORGANIZATION_MEMBER.

I have some private repositories inside an organisation which I also own, and it seems like this is not included in the tonp languages overview - do you know if the api can access private repos inside an organisation?

**Comment by patriciacostadacruz** [comment]
> https://api.github.com/users/Mr-jiangzhiguo/orgs

Hi Florian, I have deployed my own Vercel instance with my PAT_1 and it still does not show me my top languages from contributions in organizations to which I am either owner either member. Any idea? 

I am using the following -> URL +  **/api/top-langs/?username=patriciacostadacruz&layout=compact&role=OWNER,COLLABORATOR,ORGANIZATION_MEMBER,MEMBER&hide=html,css&theme=codeSTACKr&langs_count=8**

**Comment by Fabbzi** [comment]
> [@guillaumearnx](https://github.com/guillaumearnx) In that case, you need to create your own instance since a Personal Access Token needs to be present in the instance. I am happy to help if you run into problems. The following steps should get it working for you:
> 
> 1. Fork the `https://github.com/anuraghazra/github-readme-stats` repository.
> 2. Follow the Vercel instance [documentation](https://github.com/anuraghazra/github-readme-stats#deploy-on-your-own-vercel-instance) to set up your own Vercel instance.
> 3. Clone your fork to your PC `git clone <FORK_SSH_URL>`.
> 4. Create a new branch `git checkout -b org-stats`.
> 5. Add [@developStorm](https://github.com/developStorm) remote to your remotes `git remote add develop git@github.com:developStorm/github-readme-stats.git`.
> 6. Fetch the remote `git fetch develop`
> 7. Merge [@developStorm](https://github.com/developStorm) changes into your branch `git merge develop/master`.
> 8. Push these changes to your remote `git push`.
> 9. On Vercel you will see that an instance has been deployed for the `org-stats` branch.
> 10. You can use the link that is shown for that deployment to create your cards.

@rickstaa 
I used these instructions and there were merge conflicts so i just used developStorms files but I just get Invalid upstream response (401) and a broken stats card

**Comment by shimaenaga1123** [comment]
May I ask if there is any working method to see organization stats?

**Comment by PATTASWAMY-VISHWAK-YASASHREE** [comment]
# About Me

👋 Hi, I'm **Pattaswamy Vishwak Yashasree**! I'm a passionate developer and open-source enthusiast, currently contributing to innovative projects that make creative collaboration easier.

## 🚀 What I'm Working On

I'm actively involved in **Drawnix** - an open-source, all-in-one whiteboard application built on the **Plait drawing framework**. This powerful tool supports:
- 🧠 Mind maps and flowcharts
- 🖌️ Freehand drawing with pressure sensitivity
- 📊 Plugin-based architecture for extensibility
- 🌍 Multi-language support (English, Chinese, Russian, Arabic)
- 💾 Export to PNG, JPG, SVG, and JSON formats

## 💻 Tech Stack & Interests

- **Frontend**: React, TypeScript, SCSS
- **Architecture**: Plugin systems, state management with React Context
- **Tools**: Nx monorepo, Vitest for testing, ESLint
- **Interests**: Collaborative tools, creative software, internationalization

## 🌟 Current Focus

Building scalable, user-friendly applications with a focus on:
- Plugin architecture patterns
- Translation-first development
- Cross-platform compatibility
- Performance optimization

Feel free to explore my repositories and connect! I'm always excited to collaborate on open-source projects that push the boundaries of creative technology.

[![GitHub](https://img.shields.io/badge/GitHub-PATTASWAMY--VISHWAK--YASHASREE-blue)](https://github.com/PATTASWAMY-VISHWAK-YASHASREE)

**Comment by alexiam10** [comment]
Hello from PowerShell ??

**Comment by UlyssesZh** [comment]
The rate limiting problem can be solved after #4268.

**Comment by qwerty541** [comment]
> May I ask if there is any working method to see organization stats?

Hello @shimaenaga1123 

At the moment the only available way to include the organization data into your stats card is deploying the following branch https://github.com/anuraghazra/github-readme-stats/pull/2459 on your own instance https://github.com/anuraghazra/github-readme-stats?tab=readme-ov-file#deploy-on-your-own

**Comment by dariocurr** [comment]
Is there any solution working to allow that?

I tried to merge #2459 but is not working

**Comment by qwerty541** [comment]
> Is there any solution working to allow that?
> 
> I tried to merge [#2459](https://github.com/anuraghazra/github-readme-stats/pull/2459) but is not working

@dariocurr Hello

To make this work you can deploy the following branch on your own instance https://github.com/anuraghazra/github-readme-stats/pull/2459 and then add appropriate query string parameters to your card as show at this example https://github.com/anuraghazra/github-readme-stats/pull/2459#issuecomment-2011861153

**Comment by dariocurr** [comment]
> > Is there any solution working to allow that?
> > I tried to merge [#2459](https://github.com/anuraghazra/github-readme-stats/pull/2459) but is not working
> 
> [@dariocurr](https://github.com/dariocurr) Hello
> 
> To make this work you can deploy the following branch on your own instance [#2459](https://github.com/anuraghazra/github-readme-stats/pull/2459) and then add appropriate query string parameters to your card as show at this example [#2459 (comment)](https://github.com/anuraghazra/github-readme-stats/pull/2459#issuecomment-2011861153)

unfortunately it's not working, as you can see here @dariocurr (rust & python percentages should have increased a lot)

**Comment by aligheshlaghi97** [comment]
Hey, huge thanks to this repo for the inspiration! Since aggregating Org (if you're owner) stars wasn't working for me, I built a simple self-hosted workaround for anyone who needs it (requires vercel deployment with your PAT):
[github-stats-aggregator](https://github.com/aligheshlaghi97/github-stats-aggregator)

- Total Stars & contribution: Sums personal + organization accounts.
- Star cache: Won't show 0 if the GraphQL don't work.

**Comment by sskorol** [comment]
Claude was able to figure that out and assist with customization in 30 min. Feel free to build and customize on your own: https://github.com/sskorol/github-readme-stats/tree/feature/org-languages
Also pushed a docker image here: https://hub.docker.com/r/sskorol/github-readme-stats
You can build it on your own from here: https://github.com/sskorol/gh-stats
Now I can finally see the actual language stats, including private org repos.
Note that you need a classic PAT to make it work.

**Comment by Laurie-Lin** [comment]
I'm not sure if this issue that was raised in 2020 is really that hard to implement. Why, after so much discussion, is there still no solution? Everyone has started to fork their own repositories to implement it and submit pr. Is it really that difficult? These are all the functions everyone needs. I spent a lot of time browsing the entire current issue. It's obvious that the problem has been solved. Just merge the PR. Everyone using the main repository can also have this function. There's no need to search for your own implemented version from various comments anymore. Thank you

**Comment by dariocurr** [comment]
I ended up developing my solution: https://github.com/dariocurr/github-readme-stats/.
It counts contributions differently and tracks every contribution to every project.

Here the changes: https://github.com/anuraghazra/github-readme-stats/compare/master...dariocurr:github-readme-stats:master

**Comment by Laurie-Lin** [comment]
> I ended up developing my solution: https://github.com/dariocurr/github-readme-stats/. It counts contributions differently and tracks every contribution to every project.
> 
> Here the changes: [master...dariocurr:github-readme-stats:master](https://github.com/anuraghazra/github-readme-stats/compare/master...dariocurr:github-readme-stats:master)

Thank you. I'm about to try your version. What parameters should I use to include information such as the organization's star, and what are its rules? For example, if I participate in other people's projects, will those projects be counted as my star? I hope those won't be counted, or is there a parameter switch

**Comment by rickstaa** [comment]
For future reference this feature is already available on https://github.com/stats-organization/github-stats-extended/blob/master/docs/fork.md#aggregate-stats-across-organizations created by @martin-mfg. We are currently working on to get the project mantained again. We did contact @anuraghazra and will try to backport this eventually to this repository.
